src/Bicep.Cli/Logging/BicepDiagnosticLogger.cs

```diff
@@ -33,10 +35,25 @@ public void LogDiagnostic(Uri fileUri, IDiagnostic diagnostic, ImmutableArray<in
             this.logger.Log(ToLogLevel(diagnostic.Level), message);
 
             this.HasLoggedErrors |= diagnostic.Level == DiagnosticLevel.Error;
+
+            // Increment counters
+            if (diagnostic.Level == DiagnosticLevel.Warning) { this.WarningCount++; }
+            if (diagnostic.Level == DiagnosticLevel.Error) { this.ErrorCount++; }
+        }
+
+        public void LogSummary() 
+        {
+            var summary = $"Build {(this.HasLoggedErrors ? "failed" : "succeeded")}: {this.WarningCount} Warning(s), {this.ErrorCount} Error(s)";
+            
+            this.logger.Log(ToLogLevel(DiagnosticLevel.Info), summary);
         }
```

Comment:
The changes look good. Please could update the integration tests under https://github.com/Azure/bicep/blob/main/src/Bicep.Cli.IntegrationTests/ProgramTests.cs to validate the summary message?

Reply:
Thanks for the feedback! I have updated the tests to validate the summary messages.